src/app/spec/component-fixture-spec.helper.ts

```diff
@@ -0,0 +1,119 @@
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {
+    ChangeDetectorRef, DebugElement, SchemaMetadata, Type, Component, ViewChild,
+    ViewContainerRef, ComponentFactoryResolver, Optional, NgModule
+} from '@angular/core';
+import {SelectSpecHelper} from './select-spec.helper';
+import {Predicate} from '@angular/core/src/debug/debug_node';
+import {BrowserDynamicTestingModule} from '@angular/platform-browser-dynamic/testing';
+
+@Component({
+    selector: 'wave-test-host',
+    template: `<ng-template #component></ng-template>`
+})
+class TestHostComponent {
+    @ViewChild('component', {read: ViewContainerRef}) component: ViewContainerRef;
+
+    constructor(private componentFactoryResolver: ComponentFactoryResolver) {}
+
+    public load<T>(type: Type<T>, inputs: any): T {
+        const componentFactory = this.componentFactoryResolver.resolveComponentFactory(type);
+        this.component.clear();
+        const componentRef = this.component.createComponent(componentFactory);
+        if (inputs) {
+            for (const input in inputs) {
+                if (inputs.hasOwnProperty(input)) {
+                    componentRef.instance[input] = inputs[input];
+                }
+            }
+        }
+        return componentRef.instance;
+    };
+}
+
+export class ComponentFixtureSpecHelper<T> {
+
+    component: T;
+    fixture: ComponentFixture<TestHostComponent>;
+    cd: ChangeDetectorRef;
```

Comment:
You can write out the name of the three components. Your autocomplete does the typing for you.